### PR TITLE
Add 314 release notes

### LINKED ISSUE #879: Release notes for 314
## Dain Sundstrom       
- [ ] all checked
-  06-01 11:02:12    33d31950f9 Remove use of TestNG Factory in TestRegexpFunctions
-  06-01 11:02:12    6651ea48d8 Log hung tests
-  06-01 11:02:12    6984568218 Fail on concurent tests with the same name
-  06-01 11:02:12    de3abf748c Log what tests are executed and their duration
-  06-02 22:37:25    05837d44e0 Remove BlockEncodingManager
-  06-03 20:17:50    86fa8533aa Change cluster resource to use DispatchManager
-  06-03 20:46:11    3e312d8b72 Fix TestCostCalculator to use real table handles
-  06-04 08:10:15    ca9611c63f Compute estimated size directly from slice
-  06-05 09:56:57    22bc489593 Fix equals and hashCode of CallExpression to include return type
-  06-05 09:56:57    cfef9aba99 Map BETWEEN to (min <= value AND value <= max)
-  06-05 09:56:57    e2184526b3 Split special form of row expressions to separate node
## David Phillips       
- [x] all checked
-  06-01 15:44:50    08ce73b6ee Add applyDelete to push delete into connectors
-  06-01 15:44:50    1f03ef815c Rename MetadataDelete to TableDelete
-  06-01 15:44:50    486fe07e35 Skip fetching table metadata in HivePartitionManager
-  06-01 15:44:50    6c8b435a1d Use JsonCodec in RecordingHiveMetastore
-  06-01 15:44:50    7507e1da39 Cleanup error handling for Hive analyze
-  06-01 15:44:50    80a2c5113d Improve naming of WriterTarget subclasses
-  06-01 15:44:50    a389636715 Remove table layouts from Hive
-  06-01 15:44:50    a42395190c Remove redundant TestJsonHiveHandles
-  06-01 15:44:50    c327ce07ef Move getPartitionsAsList to HivePartitionManager
-  06-01 15:44:50    d821f0ec0e Remove redundant Hive test
-  06-01 15:44:50    eb5e65a977 Deprecate table layouts for makeCompatiblePartitioning
-  06-01 15:44:50    f73818870a Cleanup error message for unexpected Hive table name
-  06-02 18:27:06    45b4534d96 Fix typo in TestTsvPrinter
-  06-02 18:27:06    daf5ce670e Support JSON CLI output format
-  06-03 09:42:58    0243f01081 Disable insert into Hive table with skip property
-  06-03 09:42:58    6f2e05c265 Fix typo in TestInsertIntoHiveTable
-  06-03 10:10:19    4a23d7fad8 Minimize size of Docker image
-  06-05 10:11:35    2c8a654eac Allow connectors without distributed tables
-  06-05 10:11:35    baa87273d4 Cleanup formatting for empty methods in Connector
-  06-05 10:11:35    e5cb69e5b3 Do not return ConnectorRecordSetProvider from Kudu
-  06-05 12:56:37    bec0121e1e Cleanup Hive testing shell scripts
-  06-05 12:56:37    c5880c6481 Add script to start Hive in Docker
-  06-05 14:35:30    d6edbec7af Simplify ClientTypeSignature#toString
-  06-05 14:37:00    a4d13428ec Support skip header and footer for hive CREATE TABLE
-  06-05 15:33:16    066cf5c2c8 Simplify code in InternalHiveSplitFactory
-  06-05 15:33:16    bf868c9c1b Do not return localhost as split location from Hive connector
## Grzegorz Kokosiński 
- [ ] all checked
-   06-06 01:19:57    a4b2887909 Add WriteNullFunction to WriteMapping
-   06-06 12:09:06    4f404eee52 Add test coverage for SHOW CREATE VIEW
-   06-06 12:09:06    f1b3535a75 Code cleanup in ShowQueriesRewrite
## Karol Sobczak        
- [ ] all checked
-  06-03 02:43:01    6e1ac2098e Add WorkProcessorOperatorAdapter and use it in TopNOperatorFactory
-  06-03 02:43:01    8ae64d5ccd Do not update peak memory when allocation didn't change
-  06-03 02:43:01    ea0b6871b3 Make TopNOperatorBenchmark more stable
-  06-03 02:58:43    aec66e3ce5 Avoid unwanted Dynamic Filters on build side
-  06-03 02:58:43    b774c99cff Dynamic filtering: Planner
-  06-03 12:43:58    cd6f7a64f1 Simplify condition
## Martin Traverso      
- [x] all checked
-  06-02 07:44:17    4217df64c4 Support mixed case fields in Elasticsearch
-  06-02 14:36:24    5b411df74c Render plan structure
-  06-02 17:48:27    31c233cff6 Improve error for too many arguments to function
-  06-03 17:43:36    ddd6159ece Remove TableLayout from Local file connector
-  06-03 17:51:21    a6a09f4898 Remove TableLayout from Atop connector
-  06-03 17:56:36    7e5dded28b Remove TableLayout from Redis connector
-  06-03 17:59:08    d37eddd6c2 Remove TableLayout from Mongo connector
-  06-03 17:59:53    77c385d84a Remove TableLayout from Kafka connector
-  06-03 22:38:55    2319f940c5 Skip vertical lines when root node has no children
-  06-03 22:38:55    36aed2a505 Fix base indent level during plan rendering
-  06-04 07:03:09    85355ff713 Add support for row subscript
-  06-04 07:03:09    de75aefb62 Use helper method
-  06-04 07:38:16    1ea15426ed Remove TableLayout from Elasticsearch connector
-  06-04 07:39:17    435329db87 Remove TableLayout from Accumulo connector
-  06-04 18:49:12    11d3734546 Add Prepared statement sql to query lifecycle events
-  06-04 22:55:42    472a8f1222 Allow configuring ZSTD and LZ4 codecs for Hive compression
-  06-04 22:55:42    5564ed37af Add support for reading LZ4 and ZSTD-compresed parquet data
-  06-04 22:55:42    ecaa0db5a7 Use aircompressor ZSTD compressor for ORC
-  06-05 14:05:09    9bef5768ea Update airlift to 0.183
-  06-05 14:23:53    9183b951c8 Do not process Apply assignments via expression rewrites
## Piotr Findeisen      
- [ ] all checked
-  06-01 06:24:38    9d5468081b Use QueryCardinalityUtil#isAtMost helper
-  06-04 05:17:25    d8f0650c78 Support PostgreSQL array created using array internal type name
-  06-04 08:22:40    179ffe409f Fix getting views for Hive 2.3+ metastore
-  06-05 06:29:14    3415024362 Fix Avro schema loading when SERDEPROPERTIES set
-  06-05 06:29:14    bbc9980002 fixup! Fix Avro schema loading when SERDEPROPERTIES set
-  06-06 01:59:59    11c5c3f5e5 Use /var/log for request log in RPM

### PR DESCRIPTION
**Pull request by martint**
Closes #879

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-314.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
+===========
+Release 314
+===========
+
+General Changes
+---------------
+
+* Fix queued queries counter in UI. (:issue:`894`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-314.rst]: Move to "Web UI Changes" section

File: presto-docs/src/main/sphinx/release/release-314.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+* Fix incorrect results for ``BETWEEN`` with a ``NULL`` min or max value. (:issue:`877`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-314.rst]: endpoint?

File: presto-docs/src/main/sphinx/release/release-314.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+* Fix query history leak in coordinator. (:issue:`939` :issue:`944`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-314.rst]: Add comma between issue numbers

File: presto-docs/src/main/sphinx/release/release-314.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+* Improve performance of :func:`json_parse` function. (:issue:`904`)
+* Visualize plan structure in ``EXPLAIN`` output. (:issue:`888`)
+* Add support for positional access to ``ROW`` fields via the subscript
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-314.rst]: "constant positional access"

File: presto-docs/src/main/sphinx/release/release-314.rst [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+* Fix listing tables and views from Hive 2.3+ Metastore on certain databases,
+  including Derby and Oracle. This fixes ``SHOW TABLES``, ``SHOW VIEWS`` and
+  reading from ``information_schema.tables`` table. (:issue:`833`)
+* Fix handling of Avro tables with ``avro.schema.url`` defined in Hive
+  ``SERDEPROPERTIES``. (:issue:`898`)
+* Fix regression that caused ORC bloom filters to be ignored. (:issue:`921`)
+* Add support for reading LZ4 and ZSTD compressed Parquet data. (:issue:`910`)
+* Add support for writing ZSTD compressed ORC data. (:issue:`910`)
+* Add support for configuring ZSTD and LZ4 as default compression methods via the
+  ``hive.compression-codec`` config option. (:issue:`910`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-314.rst]: configuration property

File: presto-docs/src/main/sphinx/release/release-314.rst [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+* Support PostgreSQL arrays declared using internal type
+  name, for example ``_int4`` (rather than ``int[]``). (:issue:`659`)
+
+Elasticsearch Connector Changes
+-------------------------------
+
+* Add support for mixed-case field names. (:issue:`887`)
+
+JDBC-Based Connectors Changes
+---------------------------------
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-314.rst]: Header should match text length

Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-314.rst]: This is a library change, not a user visible connector change, so the title should be
```
Base-JDBC Connector Library Changes
-----------------------------------

File: presto-docs/src/main/sphinx/release/release-314.rst [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+* Allow JDBC-Based Connector to customize how they store ``NULL`` values. (:issue:`918`)
+
+RPM
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-314.rst]: This should be "RPM Server Changes" and moved above. We used this ordering for release 307:
* General Changes
* Web UI Changes
* Server RPM Changes

File: presto-docs/src/main/sphinx/release/release-314.rst [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+---
+
+* Change default location of the ``http-request.log`` to ``/var/log/presto``. Previously,
+  the log would be located in ``/var/lib/presto/data/var/log/`` by default. (:issue:`919`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-314.rst]: Remove trailing slash in path (to be consistent with other one)

File: presto-docs/src/main/sphinx/release/release-314.rst [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+SPI Changes
+-----------
+
+* Expose the SQL text of the executed prepared statement to ``EventListener``
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-314.rst]: Remove "plugins" word (sounds strange)

File: presto-docs/src/main/sphinx/release/release-314.rst (hunk 5 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-314.rst]: We usually say "configuration option", though

File: presto-docs/src/main/sphinx/release/release-314.rst (hunk 4 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-314.rst]: "constant positional access" sounds very strange. If anything, it should be "... via the subscript operator using a constant index". But I think that's just a detail (covered in the appropriate documentation section and via error reporting) and not relevant to the release notes.

File: presto-docs/src/main/sphinx/release/release-314.rst [hunk 10]
```diff
@@ -0,0 +1,74 @@
+===========
+Release 314
+===========
+
+General Changes
+---------------
+
+* Fix incorrect results for ``BETWEEN`` with a ``NULL`` min or max value. (:issue:`877`)
+* Fix query history leak in coordinator. (:issue:`939`, :issue:`944`)
```
Comment (dain) [inline presto-docs/src/main/sphinx/release/release-314.rst]: Add this:
```
* Fix idle client timeout handling. (:issue `947`)
```

File: presto-docs/src/main/sphinx/release/release-314.rst [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+Elasticsearch Connector Changes
+-------------------------------
+
+* Add support for mixed-case field names. (:issue:`887`)
+
+Base-JDBC Connector Library Changes
+-----------------------------------
+
+* Allow JDBC-Based Connector to customize how they store ``NULL`` values. (:issue:`918`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-314.rst]: Let’s shorten to “Allow connectors to”

**Pull request closed without merging**